FlowCrypt/src/test/java/com/flowcrypt/email/security/pgp/PgpMsgTest.kt

```diff
@@ -0,0 +1,217 @@
+      return String(bytes, Charset.forName(charset))
+    }
+
+    private fun loadResource(path: String): ByteArray {
+      return PgpMsgTest::class.java.classLoader!!
+        .getResourceAsStream("${PgpMsgTest::class.simpleName}/$path")
+        .readBytes()
+    }
+
+    private fun loadResourceAsString(path: String): String {
+      return String(loadResource(path), StandardCharsets.UTF_8)
+    }
+
+    private fun loadSecretKey(file: String): PGPSecretKeyRing {
+      return PGPainless.readKeyRing().secretKeyRing(loadResourceAsString("keys/$file"))
+    }
+
+    private fun loadMessage(file: String): String {
+      return loadResourceAsString("messages/$file")
+    }
+
+    private fun findMessage(key: String): MessageInfo {
+      return messages.firstOrNull { it.key == key }
+        ?: throw IllegalArgumentException("Message '$key' not found")
+    }
+  }
+
+  @Test // ok
+  fun multipleDecryptionTest() {
+    val keys = listOf(
+      "decrypt - without a subject",
+      "decrypt - [enigmail] encrypted iso-2022-jp pgp-mime",
+      "decrypt - [enigmail] encrypted iso-2022-jp, plain text",
+      "decrypt - [gpg] signed fully armored message"
+    )
+    for (key in keys) {
+      println("Decrypt: '$key'")
+      val r = processMessage(key)
+      assertTrue("Message not returned", r.content != null)
+      val messageInfo = findMessage(key)
+      checkContent(
+        expected = messageInfo.content,
+        actual = r.content!!.toByteArray(),
+        charset = messageInfo.charset
+      )
+    }
+  }
+
+  @Test // ok
+  fun decryptionTest1() {
+    val r = processMessage("decrypt - [security] mdc - missing - error")
+    assertTrue("Message is returned when should not", r.content == null)
+    assertTrue("Error not returned", r.error != null)
+    assertTrue("Missing MDC not detected", r.error!!.type == PgpMsg.DecryptionErrorType.NO_MDC)
+  }
+
+  @Test // ok
+  fun decryptionTest2() {
+    val r = processMessage("decrypt - [security] mdc - modification detected - error")
+    assertTrue("Message is returned when should not", r.content == null)
+    assertTrue("Error not returned", r.error != null)
+    assertTrue("Bad MDC not detected", r.error!!.type == PgpMsg.DecryptionErrorType.BAD_MDC)
+  }
+
+  // TODO: Should there be any error?
+  @Test
+  fun decryptionTest3() {
+    val r = processMessage(
+      "decrypt - [everdesk] message encrypted for sub but claims " +
+          "encryptedFor-primary,sub"
+    )
+
+    assertTrue("Message not returned", r.content != null)
+    assertTrue("Error returned", r.error == null)
+  }
+
+  // TODO: missing checksum not detected (is it same as MDC?), and should there be an error at all?
+  @Test
+  fun decryptionTest4() {
+    val r = processMessage("decrypt - encrypted missing checksum")
+    assertTrue("Message not returned", r.content != null)
+    assertTrue("Error returned", r.error == null)
+  }
```

Comment:
This is a missing armor checksum - different from MDC. Usually the four digits at the and like `p3Fc=`. Such messages are still valid if this is missing, and should decrypt correctly - so it's good as is. Maybe you should add this comment into code so it doesn't confuse us later.

Reply:
Added comment